src/transformers/models/bart/modeling_bart.py

```diff
@@ -108,12 +108,12 @@ class BartLearnedPositionalEmbedding(nn.Embedding):
     This module learns positional embeddings up to a fixed maximum size.
     """
 
-    def __init__(self, num_embeddings: int, embedding_dim: int, padding_idx: int):
-        assert padding_idx is not None, "`padding_idx` should not be None, but of type int"
+    def __init__(self, num_embeddings: int, embedding_dim: int):
         # Bart is set up so that if padding_idx is specified then offset the embedding ids by 2
         # and adjust num_embeddings appropriately. Other models dont have this hack
         self.offset = 2
-        super().__init__(num_embeddings + self.offset, embedding_dim, padding_idx=padding_idx)
+        super().__init__(num_embeddings + self.offset, embedding_dim)
+        self.weight.data[: self.offset].fill_(0)
```

Comment:
This will be overwritten by the `_init_weights` in `BartPretrainedModel` which is called after the init. Not sure if this is a problem or not, but those weights won't actually be 0.

Reply:
Good point! That would make the zeroing of weights here rather redundant...

Given the current implementation of bart, position embeddings at positions less than offset will be unused by the model so any zeroing would merely be code sugar.

I can remove that bit of code, or I could modify _init_weights.

Reply:
I think, we can just remove this line then actually, it's inaccessible anyways since `self.offset` is hardcoded to 2

Reply:
Aite, remove it is.